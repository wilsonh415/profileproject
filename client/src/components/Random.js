import React from 'react';
import { Row, Col, Image } from 'react-bootstrap';
import tameimpala from '../images/albumcovers/theslowrush.jpg';
import inreturn from '../images/albumcovers/inreturn.jpg';
import travisscott from '../images/albumcovers/birdsinthetrap.jpg';
import rhcp from '../images/albumcovers/stadiumarcadium.jpg';
import Pics from './randomComponents/Pics';

 
class Random extends React.Component {
   componentDidMount() {
       
   }
 
   render() {
       const favAlbumStyle = {
            fontSize: "35px", 
            fontFamily: "Georgia",
            marginLeft: "7vw",
            textAlign: "left",
            color: "white"
       }
       const subtitleStyle = {
        fontSize: "16px", 
        fontFamily: "Georgia",
        marginLeft: "7vw",
        textAlign: "left",
        color: "white"
        };
        const rowStyle = {
            top: "3vh",
            position: "relative",
            marginLeft: "0vw",
            fontFamily: "Georgia"
        };
       const tameStyles = {
           top: "2vh",
           position: "relative"
       }
       const odeszaStyles = {
            top: "2vh",
            position: "relative"
       }
       const textStyle = {
           fontSize: "19px",
           color: "white",
           backgroundColor: "grey",
           margin: "3vh auto",
           width: "270px"
       }
       const odeszaText = {
            marginTop: "3vh",
            fontSize: "19px",
            color: "white",
            backgroundColor: "grey",
            width: "250px",
            margin: "3vh auto"
       }
       return (
           <div className="randomStyle">
               <h3 style={favAlbumStyle}><b>Favorite Albums</b></h3>
               <h6 style={subtitleStyle}>Click on album cover to listen on Spotify!</h6>
               <div>
                   <Row style={rowStyle}>
                       <Col md={5} style={odeszaStyles}>
                        <a href="https://open.spotify.com/album/5SXT6dwhHX56Sos7KMcMF5" 
                        target="_blank" rel="noopener noreferrer">
                            <Image src={inreturn} height={220} width={220}/>
                        </a>
                       <b><p style={odeszaText}>ODESZA - In Return</p></b>
                       </Col>
                       <Col md={4} style={tameStyles}>
                        <a href="https://open.spotify.com/album/31qVWUdRrlb8thMvts0yYL" 
                        target="_blank" rel="noopener noreferrer">
                            <Image src={tameimpala} height={220} width={220}/>
                        </a>
                        <b><p style={textStyle}>Tame Impala - The Slow Rush</p></b>
                       </Col>
                   </Row>
                   <br/>
                   <Row style={rowStyle}>
                       <Col md={5} style={tameStyles}>
                        <a href="https://open.spotify.com/album/42WVQWuf1teDysXiOupIZt" 
                        target="_blank" rel="noopener noreferrer">
                            <Image src={travisscott} height={220} width={220}/>
                        </a>
                        <b><p style={textStyle}>Travis Scott - Birds in the Trap Sing McKnight</p></b>
                       </Col>
                       <Col md={4} style={odeszaStyles}>
                        <a href="https://open.spotify.com/album/7xl50xr9NDkd3i2kBbzsNZ"
                        target="_blank" rel="noopener noreferrer">
                            <Image src={rhcp} height={220} width={220}/>
                        </a>
                       <b><p style={textStyle}>Red Hot Chili Peppers - Stadium Arcadium</p></b>
                       </Col>
                   </Row>
               </div>
               <Pics></Pics>
           </div>
       );
   }
}
 
 
export default Random;
