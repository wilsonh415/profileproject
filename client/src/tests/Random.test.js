import React from 'react';
import { shallow } from 'enzyme';
import { configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import Random from '../components/Random';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

test('Random page renders correctly', () => {
    const component = renderer.create(<Random></Random>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});