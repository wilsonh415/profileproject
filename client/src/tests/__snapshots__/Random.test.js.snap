// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Random page rendered properly 1`] = `
<div
  className="randomStyle"
>
  <h3
    style={
      Object {
        "color": "white",
        "fontFamily": "Georgia",
        "fontSize": "35px",
        "marginLeft": "7vw",
        "textAlign": "left",
      }
    }
  >
    <b>
      Favorite Albums
    </b>
  </h3>
  <h6
    style={
      Object {
        "color": "white",
        "fontFamily": "Georgia",
        "fontSize": "16px",
        "marginLeft": "7vw",
        "textAlign": "left",
      }
    }
  >
    Click on album cover to listen on Spotify!
  </h6>
  <div>
    <div
      className="row"
      style={
        Object {
          "fontFamily": "Georgia",
          "marginLeft": "0vw",
          "position": "relative",
          "top": "3vh",
        }
      }
    >
      <div
        className="col-md-5"
        style={
          Object {
            "position": "relative",
            "top": "2vh",
          }
        }
      >
        <a
          href="https://open.spotify.com/album/5SXT6dwhHX56Sos7KMcMF5"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            className=""
            height={220}
            src="inreturn.jpg"
            width={220}
          />
        </a>
        <b>
          <p
            style={
              Object {
                "backgroundColor": "grey",
                "color": "white",
                "fontSize": "19px",
                "margin": "3vh auto",
                "marginTop": "3vh",
                "width": "250px",
              }
            }
          >
            ODESZA - In Return
          </p>
        </b>
      </div>
      <div
        className="col-md-4"
        style={
          Object {
            "position": "relative",
            "top": "2vh",
          }
        }
      >
        <a
          href="https://open.spotify.com/album/31qVWUdRrlb8thMvts0yYL"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            className=""
            height={220}
            src="theslowrush.jpg"
            width={220}
          />
        </a>
        <b>
          <p
            style={
              Object {
                "backgroundColor": "grey",
                "color": "white",
                "fontSize": "19px",
                "margin": "3vh auto",
                "width": "270px",
              }
            }
          >
            Tame Impala - The Slow Rush
          </p>
        </b>
      </div>
    </div>
    <br />
    <div
      className="row"
      style={
        Object {
          "fontFamily": "Georgia",
          "marginLeft": "0vw",
          "position": "relative",
          "top": "3vh",
        }
      }
    >
      <div
        className="col-md-5"
        style={
          Object {
            "position": "relative",
            "top": "2vh",
          }
        }
      >
        <a
          href="https://open.spotify.com/album/42WVQWuf1teDysXiOupIZt"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            className=""
            height={220}
            src="birdsinthetrap.jpg"
            width={220}
          />
        </a>
        <b>
          <p
            style={
              Object {
                "backgroundColor": "grey",
                "color": "white",
                "fontSize": "19px",
                "margin": "3vh auto",
                "width": "270px",
              }
            }
          >
            Travis Scott - Birds in the Trap Sing McKnight
          </p>
        </b>
      </div>
      <div
        className="col-md-4"
        style={
          Object {
            "position": "relative",
            "top": "2vh",
          }
        }
      >
        <a
          href="https://open.spotify.com/album/7Kmmw7Z5D2UD5MVwdm10sT"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            className=""
            height={220}
            src="torches.jpg"
            width={220}
          />
        </a>
        <b>
          <p
            style={
              Object {
                "backgroundColor": "grey",
                "color": "white",
                "fontSize": "19px",
                "margin": "3vh auto",
                "width": "270px",
              }
            }
          >
            Foster the People - Torches
          </p>
        </b>
      </div>
    </div>
  </div>
  <div>
    <h3
      style={
        Object {
          "color": "white",
          "fontFamily": "Georgia",
          "fontSize": "35px",
          "marginLeft": "7vw",
          "marginTop": "10vh",
          "textAlign": "left",
        }
      }
    >
      <b>
        My Pictures
      </b>
    </h3>
    <h6
      style={
        Object {
          "color": "white",
          "fontFamily": "Georgia",
          "fontSize": "16px",
          "marginLeft": "7vw",
          "textAlign": "left",
        }
      }
    >
      Click on photos to expand and for specific location!
    </h6>
    <div
      className="row"
      style={
        Object {
          "fontFamily": "Georgia",
          "marginLeft": "0vw",
          "position": "relative",
          "top": "3vh",
        }
      }
    >
      <div
        className="col-md-5"
        style={
          Object {
            "marginLeft": "4vw",
            "position": "relative",
          }
        }
      >
        <img
          className=""
          height={110}
          onClick={[Function]}
          src="LandsEnd.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={110}
        />
        <img
          className=""
          height={110}
          onClick={[Function]}
          src="SutroBaths.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={110}
        />
        <img
          className=""
          height={110}
          onClick={[Function]}
          src="PalaceOfFineArts.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={110}
        />
        <img
          className=""
          height={110}
          onClick={[Function]}
          src="InaCoolbrithPark.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={110}
        />
        <img
          className=""
          height={110}
          onClick={[Function]}
          src="LakeMerced.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={110}
        />
        <img
          className=""
          height={110}
          onClick={[Function]}
          src="DoloresPark.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={110}
        />
        <img
          className=""
          height={110}
          onClick={[Function]}
          src="CrissyFields.jpg"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={110}
        />
        <img
          className=""
          height={110}
          onClick={[Function]}
          src="GoldenGatePark.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={110}
        />
        <img
          className=""
          height={110}
          onClick={[Function]}
          src="TwinPeaks.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={110}
        />
        <b>
          <p
            style={
              Object {
                "backgroundColor": "grey",
                "color": "white",
                "fontSize": "19px",
                "margin": "3vh auto",
                "position": "relative",
                "width": "270px",
              }
            }
          >
            My Hometown - San Francisco
          </p>
        </b>
      </div>
      <div
        className="col-md-5"
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <img
          className=""
          height={165}
          onClick={[Function]}
          src="OldFaithful.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={165}
        />
        <img
          className=""
          height={165}
          onClick={[Function]}
          src="BoysenStatePark.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={165}
        />
        <img
          className=""
          height={165}
          onClick={[Function]}
          src="Yellowstone.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={165}
        />
        <img
          className=""
          height={165}
          onClick={[Function]}
          src="GrandTeton.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={165}
        />
        <b>
          <p
            style={
              Object {
                "backgroundColor": "grey",
                "color": "white",
                "fontSize": "19px",
                "margin": "3vh auto",
                "position": "relative",
                "width": "270px",
              }
            }
          >
            Wyoming
          </p>
        </b>
      </div>
    </div>
    <div
      className="row"
      style={
        Object {
          "fontFamily": "Georgia",
          "marginLeft": "0vw",
          "position": "relative",
          "top": "3vh",
        }
      }
    >
      <div
        className="col-md-5"
        style={
          Object {
            "marginLeft": "4vw",
            "position": "relative",
          }
        }
      >
        <img
          className=""
          height={165}
          onClick={[Function]}
          src="BadlandsInterior.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={165}
        />
        <img
          className=""
          height={165}
          onClick={[Function]}
          src="TheBadlands.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={165}
        />
        <img
          className=""
          height={165}
          onClick={[Function]}
          src="PioneerAutoMuseum.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={165}
        />
        <img
          className=""
          height={165}
          onClick={[Function]}
          src="CusterStatePark.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={165}
        />
        <b>
          <p
            style={
              Object {
                "backgroundColor": "grey",
                "color": "white",
                "fontSize": "19px",
                "margin": "3vh auto",
                "position": "relative",
                "width": "270px",
              }
            }
          >
            South Dakota
          </p>
        </b>
      </div>
      <div
        className="col-md-5"
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <img
          className=""
          height={165}
          onClick={[Function]}
          src="CentralPark.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={165}
        />
        <img
          className=""
          height={165}
          onClick={[Function]}
          src="TimesSquare.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={165}
        />
        <img
          className=""
          height={165}
          onClick={[Function]}
          src="BrooklynBridge.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={165}
        />
        <img
          className=""
          height={165}
          onClick={[Function]}
          src="Tarrytown.JPG"
          style={
            Object {
              "border": "2px solid white",
            }
          }
          width={165}
        />
        <b>
          <p
            style={
              Object {
                "backgroundColor": "grey",
                "color": "white",
                "fontSize": "19px",
                "margin": "3vh auto",
                "position": "relative",
                "width": "270px",
              }
            }
          >
            New York
          </p>
        </b>
      </div>
    </div>
  </div>
</div>
`;
